analysis/monk/src/DampenHarm.tsx

```diff
@@ -0,0 +1,80 @@
+import { formatNumber, formatThousands } from 'common/format';
+import SPELLS from 'common/SPELLS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, {
+  DamageEvent,
+  ResourceChangeEvent,
+  DrainEvent,
+  HealEvent,
+} from 'parser/core/Events';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+
+class DampenHarm extends Analyzer {
+  hitsReduced = 0;
+  damageReduced = 0;
+  currentMaxHP = 0;
+
+  constructor(options: Options) {
+    super(options);
+    if (!this.selectedCombatant.hasTalent(SPELLS.DAMPEN_HARM_TALENT.id)) {
+      this.active = false;
+      return;
+    }
+    this.addEventListener(Events.damage.to(SELECTED_PLAYER), this.updateMaxHP);
+    this.addEventListener(Events.resourcechange.by(SELECTED_PLAYER), this.updateMaxHP);
+    this.addEventListener(Events.drain.by(SELECTED_PLAYER), this.updateMaxHP);
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER), this.updateMaxHP);
```

Comment:
iirc this should be .to not .by
As the maxHP of the event would be of the target not the source (which while you're not doing a ton of external healing it can mess things up) 
This way you will also capture the maxHP due to things like druid hots, paladin beacon etc etc
More events with the data

Reply:
Drain also needed to be swapped to "to", which I also did. You know, in case you run WoWA on a TBC dungeon which still has mana burn and stuff...?